openpype/hosts/maya/plugins/publish/extract_arnold_scene_source.py

```diff
@@ -113,34 +111,63 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
     def _extract(self, nodes, attribute_data, kwargs):
-        self.log.info("Writing: " + kwargs["filename"])
+        self.log.info(
+            "Writing {} with:\n{}".format(kwargs["filename"], kwargs)
+        )
         filenames = []
         # Duplicating nodes so they are direct children of the world. This
         # makes the hierarchy of any exported ass file the same.
-        with delete_after() as delete_bin:
+        with lib.delete_after() as delete_bin:
             duplicate_nodes = []
             for node in nodes:
+                # Only interested in transforms:
+                if cmds.nodeType(node) != "transform":
+                    continue
+
+                # Only interested in transforms with shapes.
+                shapes = cmds.listRelatives(
+                    node, shapes=True, fullPath=True
+                ) or []
+                if not shapes:
+                    continue
+
+                parent = cmds.listRelatives(
+                    node, parent=True, fullPath=True
+                )[0]
                 duplicate_transform = cmds.duplicate(node)[0]
+                duplicate_transform = "{}|{}".format(
+                    parent, duplicate_transform
+                )
+
+                if cmds.listRelatives(duplicate_transform, parent=True):
+                    duplicate_transform = cmds.parent(
+                        duplicate_transform, world=True
+                    )[0]
+
+                basename = node.split("|")[-1].split(":")[-1]
+                duplicate_transform = cmds.rename(
+                    duplicate_transform, basename
+                )
 
-                # Discard the children.
-                shapes = cmds.listRelatives(duplicate_transform, shapes=True)
+                # Discard children nodes that are not shapes
+                shapes = cmds.listRelatives(
+                    duplicate_transform, shapes=True, fullPath=True
+                )
                 children = cmds.listRelatives(
-                    duplicate_transform, children=True
+                    duplicate_transform, children=True, fullPath=True
                 )
                 cmds.delete(set(children) - set(shapes))
 
-                duplicate_transform = cmds.parent(
-                    duplicate_transform, world=True
-                )[0]
-
-                cmds.rename(duplicate_transform, node.split("|")[-1])
-                duplicate_transform = "|" + node.split("|")[-1]
-
                 duplicate_nodes.append(duplicate_transform)
+                duplicate_nodes.extend(shapes)
```

Comment:
Aren't the shapes automatically deleted since they are children of the transform?

Reply:
The `duplicate_nodes` variable is for adding `cbId`. Not for deletion.